c2corg_ui/static/js/search/pagination.js

```diff
@@ -85,11 +91,10 @@ app.PaginationController.prototype.handleSearchChange_ = function(event,
     features, total, recenter) {
   this.total = total;
   this.offset = this.location_.getFragmentParamAsInt('offset') || 0;
-  // don't show the "Go to last page" button when doing a full-text-search
-  // (the full result set has to be iterated in ElasticSearch when doing
-  // requests with large offsets)
-  this.showGoToLastPage = !(this.location_.hasFragmentParam('q') &&
-    this.location_.getFragmentParam('q') !== '');
+  // don't show the "Go to last page" button if the offset is above
+  // maxResultOffset_, or the API will return an error
+  var lastOffset = this.total - (this.total % this.limit);
+  this.showGoToLastPage = lastOffset <= this.maxResultOffset_ - this.limit;
```

Comment:
Wouldn't `this.showGoToLastPage = this.total < this.maxResultOffset_` be enough?